src/service/resources-impl.js

```diff
@@ -1298,17 +1292,20 @@ export class ResourcesImpl {
       // that ignores `relayoutTop` and `elementsThatScrolled`.
       for (let i = 0; i < this.resources_.length; i++) {
         const r = this.resources_[i];
-        if (r.hasOwner()) {
+        const requested = r.isMeasureRequested();
+        if (r.hasOwner() && !requested) {
           continue;
         }
-        // Difference: if we have a "premeasured" client rect, consume it
-        // as the element's new measurements even if the element isn't built.
-        const requested = r.isMeasureRequested();
+        const premeasured = r.hasBeenPremeasured();
         if (requested) {
           dev().fine(TAG_, 'force remeasure:', r.debugid);
         }
-        const premeasured = r.hasBeenPremeasured();
-        const needsMeasure = premeasured || requested || this.relayoutAll_;
+        // Immediately measure (vs. waiting for async premeasure) for certain
+        // elements with sensitive time-to-measure.
+        const expediteFirstMeasure =
+          !r.hasBeenMeasured() && r.element.tagName == 'AMP-AD';
+        const needsMeasure =
+          premeasured || requested || this.relayoutAll_ || expediteFirstMeasure;
```

Comment:
FYI these changes bring the experiment path closer to the non-experiment path: https://github.com/ampproject/amphtml/pull/29581/files#diff-80b9cea934a00a64f15d14cb656dfad4R1339-R1344